server/nodeset/upload-deposit-data.go

```diff
@@ -132,25 +128,21 @@ func (c *nodesetUploadDepositDataContext) PrepareData(data *swapi.NodesetUploadD
 	}
 	data.Balance = balance
 
-	totalCost := new(big.Int)
+	totalCost := big.NewInt(0)
+	costPerKey := eth.EthToWei(validatorDepositCost)
 
-	// Add deposit cost for each unregistered key
-	for range unregisteredKeys {
-		totalCost.Add(totalCost, eth.EthToWei(validatorDepositCost))
-	}
-	// Total cost needs to account for pending validators
-	for range pendingPubkeysOnNodeset {
-		totalCost = totalCost.Add(totalCost, eth.EthToWei(validatorDepositCost))
-	}
+	unregisteredKeysCount := len(unregisteredKeys)
+	pendingPubkeysOnNodesetCount := len(pendingPubkeysOnNodeset)
+
+	totalCost.Add(totalCost, new(big.Int).Mul(costPerKey, big.NewInt(int64(unregisteredKeysCount+pendingPubkeysOnNodesetCount))))
 
 	data.SufficientBalance = (totalCost.Cmp(balance) <= 0)
 
 	if !data.SufficientBalance {
-		for len(unregisteredKeys) > 0 && totalCost.Cmp(balance) > 0 {
+		for totalCost.Cmp(balance) > 0 {
 			unregisteredKeys = unregisteredKeys[1:]
 			unregisteredPubkeys = unregisteredPubkeys[1:]
-			totalCost = totalCost.Sub(totalCost, eth.EthToWei(validatorDepositCost))
-
+			totalCost = totalCost.Sub(totalCost, costPerKey)
```

Comment:
You can actually just do `totalCost.Sub(totalCost, costPerKey)` without an assignment and it'll assign for you. Big math is kind of weird like that.

Reply:
resolved